x-pack/plugins/secops/public/lib/helpers/index.test.tsx

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { decodeIpv6, encodeIpv6 } from '../helpers';
+
+describe('Helpers', () => {
+  describe('#encodeIpv6', () => {
+    test('if it encodes the provided IPv6 by replacing : with -', () => {
+      const encodedIp = encodeIpv6('2001:db8:ffff:ffff:ffff:ffff:ffff:ffff');
+      expect(encodedIp).toBe('2001-db8-ffff-ffff-ffff-ffff-ffff-ffff');
+    });
+  });
+
```

Comment:
Do you need to protect against zones? Might be better to be safe than sorry in case we get those showing up?

```ts
    // https://en.wikipedia.org/wiki/IPv6_address#Use_of_zone_indices_in_URIs
    test('if it encodes the provided IPv6 with zones by replacing % with -', () => {
      const encodedIp = encodeIpv6('fe80:3438:7667:5c77:ce27%18');
      expect(encodedIp).toBe('fe80-3438-7667-5c77-ce27-18');
    });
```

Reply:
So looks like it doesn’t break routing. An `IPDetails` link with a zone works fine (doesn't break the back button either), but the zone isn’t part of the URL after the `link-to` re-route, so I'll look into that. Thanks for pointing this out! :)

For reference, the following:
``` javascript
<IPDetailsLink ip="fe80:3438:7667:5c77:ce27%18" />
```
Creates a link with `href`: 
`/app/secops#/link-to/network/ip/fe80-3438-7667-5c77-ce27%2518`
And the final URL in the address bar after navigating:
`/app/secops#/network/ip/fe80-3438-7667-5c77-ce27?_g=()`